### PR TITLE
Analytics: include products without orders in the reporting.

### PR DESCRIPTION
**Pull request by AnnaMag**
### All Submissions:

-   [ ] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [ ] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

Closes # .

- [ ] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

### How to test the changes in this Pull Request:

1.
2.
3.

### Other information:

-   [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [ ] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce/src/Admin/API/Reports/Variations/DataStore.php [hunk 1]
```diff
@@ -482,6 +486,13 @@ public function get_data( $query_args ) {
 			);
 			/* phpcs:enable */
 
+			$variations_query_all = "\n\t\t\tSELECT\n\t\t\t\tall_products.product_id, '' as variation_id, '' as items_sold, '' as net_revenue, '' as orders_count FROM wp_wc_product_meta_lookup as all_products LEFT JOIN {$table_name} ON {$table_name}.product_id = all_products.product_id WHERE {$table_name}.product_id IS NULL";
+			$product_data_all = $wpdb->get_results(
+				$variations_query_all,
+				ARRAY_A
+			);
+			$product_data_all_sub = array_slice($product_data_all, 0, 5);
```
Comment (AnnaMag) [inline plugins/woocommerce/src/Admin/API/Reports/Variations/DataStore.php:494]: select a few products.

File: plugins/woocommerce/src/Admin/API/Reports/Variations/DataStore.php [hunk 2]
```diff
@@ -491,7 +502,7 @@ public function get_data( $query_args ) {
 			if ( $query_args['extended_info'] ) {
 				$this->fill_deleted_product_name( $product_data );
 			}
-
+			$product_data = array_merge($product_data,$product_data_all_sub);
```
Comment (AnnaMag) [inline plugins/woocommerce/src/Admin/API/Reports/Variations/DataStore.php:505]: Paste the data together should we want to use a single table. Alternatively, we could add a separate table and paste the data on report Download.

**Comment by samueljseay** [comment]
Hi @AnnaMag 

We are closing this pull request, as the last activity has been quite some time ago and it remains in a draft state. If you are still interested or you feel like this is a mistake, please open another PR based on a recent version of the trunk branch. Thanks!

**Comment by AnnaMag** [comment]
Thanks for closing it @samueljseay! It was meant to be a POC implementation, not destined for merging.

**Pull request closed without merging**